src/prefect/engine/task_runner.py

```diff
@@ -670,6 +673,9 @@ def check_target(self, state: State, inputs: Dict[str, Result]) -> State:
         target = self.task.target
 
         if result and target:
+            if not isinstance(target, str):
+                target = target(**prefect.context)
```

Comment:
piggy backing off my comment above, there are more values in context here than when the task runner is initialized -- this call works 👍